### PR TITLE
Added Webpack as part of build process to catch issues like in https:…

### PR DESCRIPTION
**Pull request by kichalla**
…//github.com/aspnet/jquery-validation-unobtrusive/issues/79

### CODE REVIEW

File: build.msbuild [hunk 1]
```diff
@@ -13,5 +15,12 @@
         <Exec Command="npm version --no-git-tag-version --allow-same-version $(PackageVersion)" />
         <Exec Command="gulp" />
         <Exec Command="nuget pack $(NuspecPath) -Version $(PackageVersion)" />
+        <Message Text="Testing Webpack build..." />
+        <MakeDir Directories="$(NpmPackOutputDir)" />
+        <Exec Command="npm pack $(MSBuildThisFileDirectory)" WorkingDirectory="$(NpmPackOutputDir)" />
+        <Exec Command="npm init -y" WorkingDirectory="$(WebPackTestDir)" />
+        <Exec Command="npm install $(NpmPackOutputDir)jquery-validation-unobtrusive-$(PackageVersion).tgz --save-dev --force" WorkingDirectory="$(WebPackTestDir)" />
+        <Exec Command="npm install webpack webpack-cli --save-dev" WorkingDirectory="$(WebPackTestDir)" />
+        <Exec Command="npx webpack" WorkingDirectory="$(WebPackTestDir)" />
```
Comment (kichalla) [inline build.msbuild]: With the changes in this PR and the current scripts in the repo, I am able to repro the error.

@Eilon could we have this repo run in AppVeyor and Travis? (probably AppVeyor alone should be enough)

Comment (xt0rted) [reply build.msbuild]: FWIW AppVeyor has support for [linux builds](https://www.appveyor.com/docs/getting-started-with-appveyor-for-linux/) now which can be setup as part of a build matrix.

Comment (Eilon) [reply build.msbuild]: Templating should now be enabled w/ AppVeyor.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,6 @@
 bower_components/
 node_modules/
-*.nupkg
\ No newline at end of file
+*.nupkg
+npmpackoutput/
+/test/webpacktest/package*.json
```
Comment (Eilon) [inline .gitignore]: Why do these folders start with `/` but others don't?

File: build.msbuild [hunk 3]
```diff
@@ -13,5 +16,19 @@
...
+
+        <MSBuild Projects ="$(MSBuildProjectFullPath)"
+            Properties="PackageVersion=$(PackageVersion)"
+            Targets="Test" />
+    </Target>
+
+    <Target Name="Test">
+        <Message Text="Testing Webpack build..." />
+        <MakeDir Directories="$(NpmPackOutputDir)" />
+        <Exec Command="npm pack $(MSBuildThisFileDirectory)" WorkingDirectory="$(NpmPackOutputDir)" />
```
Comment (Eilon) [inline build.msbuild]: Do we have some way of describing that this repo needs NPM in order to build? cc @natemcmaster

File: build.msbuild (hunk 1 above)
Comment (Eilon) [reply build.msbuild]: Oops. Wrong repo.

Comment (Eilon) [reply build.msbuild]: jquery-validation-unobtrusive is now enabled in AppVeyor.

Comment (jbagga) [reply build.msbuild]: @Eilon aspnet/templating runs Appveyor already

File: build.msbuild (hunk 3 above)
Comment (natemcmaster) [reply build.msbuild]: This repo isn't using korebuild, so not really.

Comment (Eilon) [reply build.msbuild]: Ah right. I kept thinking this is the templating repo. But it is not.

**Comment by mkArtakMSFT** [comment]
@Eilon, can you please look through this one more time and see what is pending?

**Comment by Eilon** [comment]
Why is AppVeyor so sad?

> The build phase is set to "MSBuild" mode (default), but no Visual Studio project or solution files were found in the root directory. If you are not building Visual Studio project switch build mode to "Script" and provide your custom build command.

Maybe we need to add an `appveyor.yml` file to this repo to configure it to do the right kind of build? Or should we disable AppVeyor?

**Comment by kichalla** [comment]
@Eilon AppVeyor is now working for this repo. The build fails as expected.

**Pull request merged**